### PR TITLE
Migrating the Ballerina SAP Connector to Ballerina version 1.0.1

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
Fix https://github.com/wso2/ballerina-integrator/issues/191

### CODE REVIEW

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java [hunk 1]
```diff
@@ -16,55 +16,78 @@
...
+
+    private static Logger log = LoggerFactory.getLogger("ballerina");
 
-    private static final Log log = LogFactory.getLog(RFCConnectionHandler.class);
     private ResponseCallback callback;
-    private Map<String, Resource> sapResource;
-    private Context context;
+    private Map<String, AttachedFunction> sapResource;
+    private final BRuntime runtime;
+//    private Context context;
```
Comment (Maninda) [inline sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: Do we need this comment?

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java [hunk 2]
```diff
@@ -16,55 +16,78 @@
...
+
+    BapiHandlerFactory(Map<String, AttachedFunction> sapService) {
 
-    RFCConnectionHandler(Map<String, Resource> sapService, Context context) {
         this.sapResource = sapService;
         callback = new ResponseCallback();
-        this.context = context;
+//        this.context = context;
```
Comment (Maninda) [inline sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: Can we remove this comment if not required?

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java (hunk 1 above)
Comment (dulanjalidilmi) [reply sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: Fixed. Thanks.

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java (hunk 2 above)
Comment (dulanjalidilmi) [reply sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: Fixed. Thanks.

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java [hunk 3]
```diff
@@ -16,55 +16,73 @@
...
     public void handleRequest(JCoServerContext jCoServerContext, JCoFunction jCoFunction) {
+
         if (log.isDebugEnabled()) {
             log.debug("New BAPI function call received");
         }
         jCoFunction.getExportParameterList().setValue("ECHOTEXT", jCoFunction.getImportParameterList()
                 .getString("REQUTEXT"));
         String output = jCoFunction.getImportParameterList().getString("REQUTEXT");
-        SapUtils.invokeOnBapiMessage(output, sapResource, callback, context);
+        log.info("handleRequest1");
```
Comment (indikasampath2000) [inline sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: It seems to be a log added for testing. Better to delete.

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java [hunk 4]
```diff
@@ -16,55 +16,73 @@
...
+        try {
+            log.info("invokeOnBapiMessage");
```
Comment (indikasampath2000) [inline sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: This should be a meaningful DEBUG log.

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java [hunk 5]
```diff
@@ -16,55 +16,73 @@
...
  * This class handles the BAPI calls that are returned from the SAP gateway.
  * <p>
  * This class encapsulates each BAPI/RFC request received.
  * by handling  and converting them to Strings.
  * </p>
  */
-public class RFCConnectionHandler implements JCoServerFunctionHandler {
+public class BapiHandlerFactory implements JCoServerFunctionHandler {
+
+    private static Logger log = LoggerFactory.getLogger("ballerina");
```
Comment (indikasampath2000) [inline sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: `ballerina` should be constant.

File: sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java (hunk 4 above)
Comment (dulanjalidilmi) [reply sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java]: Fixed

**Pull request closed without merging**